src/pages/index.js

```diff
@@ -102,33 +197,104 @@ function renderCard(data) {
 }
 
 function getCardElement(cardData) {
-    const card = new Card(cardData, '#card-template', handleImageClick);
+    console.log(cardData);
+    const card = new Card(cardData, '#card-template', handleImageClick, handleDeleteButtonSubmit, handleCardLike, handleCardDislike);
     return card.getView();
 }
 
-function handleImageClick(cardData) {
-    popupWithImage.open(cardData);
+function showErrorMessage(err){
+    const messageElement = document.querySelector('.modal__button');
+    messageElement.textContent = `Error: ${err}`;
 }
 
 /* -------------------------------------------------------------------------- */
 /*                               Event Handlers                               */
 /* -------------------------------------------------------------------------- */
 
+function handleImageClick(cardData) {
+    popupWithImage.open(cardData);
+}
+
+function handleCardLike(card){
+    api
+        .likeCard(card._id)
+        .then(() => {
+            card.cardIsLiked(true);
+        })
+        .catch((err) => {
+            console.log(err);
+        });
+}
+
+function handleCardDislike(card) {
+    api
+        .dislikeCard(card._id)
+        .then(() => {
+            card.cardIsLiked(false);
+        })
+        .catch((err) => {
+            console.error(err);
+        });
+}
+
 function handleProfileEditSumbit (userData) {
-    console.log(userData);
     const name = userData.name;
     const job = userData.bio;
-    userInfo.setUserInfo(name, job);
-    popupEditForm.close();
+    console.log(name);
+    console.log(job);
+    editPostLoading.editPost();
+    api.updateProfileInfo(name, job)
+        .then(() => {
+            return api.getUserInfo();
```

Comment:
`api.getUserInfo();` is not needed here. 

You get the whole profile data in the response from the server anyway